### PR TITLE
[pcre] Export cmake targets

### PR DESCRIPTION
**Pull request by PhoebeHui**
Export cmake targets for pcre

Related to https://github.com/microsoft/vcpkg/pull/10326#discussion_r483215122

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: ports/pcre/export-cmake-targets.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -885,11 +885,19 @@ ENDIF(PCRE_BUILD_TESTS)
+ # Installation
+ SET(CMAKE_INSTALL_ALWAYS 1)
+ 
+-INSTALL(TARGETS ${targets}
++foreach(target ${targets})
++    INSTALL(TARGETS ${target}
++        EXPORT ${target}-targets
```
Comment (ras0219-msft) [inline ports/pcre/export-cmake-targets.patch:12]:
```suggestion
+        EXPORT pcre-targets
```

File: ports/pcre/export-cmake-targets.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+         RUNTIME DESTINATION bin
+         LIBRARY DESTINATION lib
+         ARCHIVE DESTINATION lib)
+ 
++    INSTALL(EXPORT ${target}-targets
```
Comment (ras0219-msft) [inline ports/pcre/export-cmake-targets.patch:17]: Move this export out of the `foreach()` loop and install `pcre-targets` once.

File: ports/pcre/unofficial-pcre-config.cmake [hunk 3]
```diff
@@ -0,0 +1,19 @@
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre-targets.cmake")
```
Comment (ras0219-msft) [inline ports/pcre/unofficial-pcre-config.cmake:1]: With the changes above, this only needs to include `unofficial-pcre-targets.cmake`

File: ports/pcre/unofficial-pcre-config.cmake [hunk 4]
```diff
@@ -0,0 +1,19 @@
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre16-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcre32-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcrecpp-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-pcreposix-targets.cmake")
+
+set(PCRE_INCLUDE_DIR ${CMAKE_CURRENT_LIST_FILE}/../../include)
```
Comment (ras0219-msft) [inline ports/pcre/unofficial-pcre-config.cmake:7]: We do not need to provide these variables; instead we should encourage users to use the targets directly

File: ports/pcre/usage [hunk 5]
```diff
@@ -0,0 +1,14 @@
+The package pcre provides CMake targets:
```
Comment (ras0219-msft) [inline ports/pcre/usage:1]: This file should be removed and we should ensure the automatic cmake guidance is sufficient

File: ports/pcre/export-cmake-targets.patch [hunk 6]
```diff
@@ -0,0 +1,25 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 252de6e..2fea7ca 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -885,11 +885,19 @@ ENDIF(PCRE_BUILD_TESTS)
+ # Installation
+ SET(CMAKE_INSTALL_ALWAYS 1)
+ 
+-INSTALL(TARGETS ${targets}
++foreach(target ${targets})
```
Comment (ras0219-msft) [inline ports/pcre/export-cmake-targets.patch:10]: These targets may need to have
```cmake
target_include_directories(${target} PUBLIC $<INSTALL_INTERFACE:include>)
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/pcre/export-cmake-targets.patch (hunk 1 above)
Comment (PhoebeHui) [reply ports/pcre/export-cmake-targets.patch]: Updated, thanks for your guidlines!

File: ports/pcre/export-cmake-targets.patch (hunk 2 above)
Comment (PhoebeHui) [reply ports/pcre/export-cmake-targets.patch]: Updated, thanks!

File: ports/pcre/unofficial-pcre-config.cmake (hunk 3 above)
Comment (PhoebeHui) [reply ports/pcre/unofficial-pcre-config.cmake:1]: Fixed, thanks!

File: ports/pcre/unofficial-pcre-config.cmake (hunk 4 above)
Comment (PhoebeHui) [reply ports/pcre/unofficial-pcre-config.cmake]: Updated, thanks for your review!

File: ports/pcre/usage (hunk 5 above)
Comment (PhoebeHui) [reply ports/pcre/usage]: Removed, thanks for pointing out this!

File: ports/pcre/export-cmake-targets.patch (hunk 6 above)
Comment (PhoebeHui) [reply ports/pcre/export-cmake-targets.patch:10]: Updated, thanks!

**Review by ras0219 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**